xwiki-rendering-macros/xwiki-rendering-macro-toc/src/main/java/org/xwiki/rendering/internal/macro/toc/TocBlockFilter.java

```diff
@@ -51,6 +52,12 @@ public TocBlockFilter(Parser plainTextParser, LinkLabelGenerator linkLabelGenera
      */
     public List<Block> generateLabel(HeaderBlock headerBlock)
     {
-        return headerBlock.clone(this).getChildren();
+        List<Block> children = headerBlock.clone(this).getChildren();
+        if (children.isEmpty()) {
+            // In case of empty header, explicitly assign an empty label. Otherwise, the toc entry is generated with 
+            // the name of the page.
+            children = List.of(new WordBlock(""));
```

Comment:
This seems like a bad hack imho. I think a `Wordblock` should be non-empty. Also, with this hack, the generated links won't be clickable in any way. I see two options:

* Use a space (i.e., `SpaceBlock`). While not really nice, it would at least ensure that you can actually click on the link I think.
* Omit empty headings in the table of contents.

Reply:
Thanks, https://forum.xwiki.org/t/behavior-of-the-table-of-content-in-case-of-empty-header/11977 created to discuss this.